moped-editor/src/components/GridTable/GridTableSearch.js

```diff
@@ -16,65 +53,311 @@ import GridTableExport from "./GridTableExport";
+    searchState.setSearchParameters({
+      value: "",
+      column: "",
+    });
+  }
+
+  /**
+   * Handles the click on a tab to switch between tabs
+   * @param {Object} event - The tab being clicked event
+   * @param {number} newTabValue - The clicked tab value
+   */
+  const handleTabChange = (event, newTabValue) => {
+    setTabValue(newTabValue);
+  };
+
+  /**
+   * Update the export whenever limit or selectall change
+   */
+  useEffect(
+    () => {
+      if (dialogOpen && !downloading) {
+        query.limit = 0;
+        getExport();
+        setDownloading(true);
+      }
+
+      if (dialogOpen && downloading && data && !loading) {
+        const formattedData = formatExportData(data[query.table]);
+        const headers = dataGetHeaders(formattedData);
+        const csvString = dataToCSV(headers, formattedData);
+        setTimeout(() => {
+          // Update the state
+          setDialogOpen(false);
+          downloadFile(csvString);
+        }, 1500);
+      }
+    },
+    // eslint-disable-next-line
+    [dialogOpen, downloading, loading, query.limit, getExport]
+  );
 
   return (
     <div>
-      <GridTableExport
-        query={query}
-        showFilterState={{
-          showFilters: showFilters,
-          setShowFilters: setShowFilters,
-        }}
-      />
+      <GridTableExport query={query} />
       {children}
+
       <Box mt={3}>
-        <Card>
-          <CardContent>
-            {/* Search-bar component */}
-            <Collapse in={defaultShowSearchBarValue && !showFilters}>
-              <GridTableSearchBar
+        <Paper>
+          <Grid container lg={12}>
+            <Grid item xs={12} sm={6} md={8} lg={10} xl={10}>
+              <Tabs
+                value={tabValue}
+                onChange={handleTabChange}
+                indicatorColor="primary"
+                textColor="primary"
+                aria-label="Search Data Table"
+              >
+                <Tab
+                  className={classes.tabStyle}
+                  onClick={handleSwitchToSearch}
+                  label={
+                    <div>
+                      <Icon style={{ verticalAlign: "middle" }}>search</Icon>{" "}
+                      Search{" "}
+                    </div>
+                  }
+                />
+                <Tab
+                  className={classes.tabStyle}
+                  onClick={handleSwitchToAdvancedSearch}
+                  label={
+                    <div>
+                      <Icon style={{ verticalAlign: "middle" }}>rule</Icon>{" "}
+                      Advanced Search{" "}
+                    </div>
+                  }
+                />
+              </Tabs>
+            </Grid>
+            <Grid
+              item
+              xs={12}
+              sm={6}
+              md={4}
+              lg={2}
+              xl={2}
+              className={classes.downloadButtonGrid}
+            >
+              {query.config.showExport && (
```

Comment:
Nice to see grid in action to keep the download button in place. I've been curious about using MUI for UI details like this, and it looks real nice along with those search tabs.